python-client/giskard/scanner/scanner.py

```diff
@@ -0,0 +1,33 @@
+from typing import Optional, Sequence
+from ..models.base import BaseModel
+from ..datasets.base import Dataset
+from ..core.model_validation import validate_model
+from .registry import DetectorRegistry
+from .result import ScanResult
+
+
+class Scanner:
+    def __init__(self, params: Optional[dict] = None):
+        self.params = params or dict()
+
+    def analyze(self, model: BaseModel, dataset: Dataset) -> ScanResult:
+        """Runs the analysis of a model and dataset, detecting issues."""
+        validate_model(model=model, validate_ds=dataset)
+
+        # Collect the detectors
+        detectors = self.get_detectors(tags=[model.meta.model_type.value])
+
+        issues = []
+        for detector in detectors:
+            issues.extend(detector.run(model, dataset))
```

Comment:
should we parallelize it already?

Reply:
Still wondering if it’s better to let the detectors manage the parallelization or to do it at this level. For now we only have a single detector so we can wait.